pystac/collection.py

```diff
@@ -305,6 +306,12 @@ class SpatialExtent:
             2D Collection with only one bbox would be [[xmin, ymin, xmax, ymax]]
     """
     def __init__(self, bboxes):
+        # A common mistake is to pass in a single bbox instead of a list of bboxes.
+        # Account for this by transforming the input in that case.
+        if isinstance(bboxes, collections.abc.Sequence):
+            if not isinstance(bboxes[0], collections.abc.Sequence):
```

Comment:
> doc changes

I see your point, but I actually don't want to encourage anyone to pass a bare list, as this is against the spec. This change is really to catch a common error.

> IndexError...raises

Yes. I also want to be light on constructor validation, because if someone is trying to read invalid STAC, I think the stance PySTAC would want to take is to be able to read it in, validate it to discover errors, and correct things in the context of PySTAC. There are some places in the codebase that we validate in the constructor, and I'm thinking that actually we should move away from that entirely for consistency, and force the user to explicitly validate if they want to check if everything is valid.